src/frontend/components/PlaylistPortability/index.tsx

```diff
@@ -0,0 +1,275 @@
+import React, { useEffect } from 'react';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+
+import { usePlaylist, useUpdatePlaylist } from '../../data/queries';
+import { ErrorMessage } from '../ErrorComponents';
+import { LTINav } from '../LTINav';
+import { Video } from '../../types/tracks';
+import { Document } from '../../types/file';
+import { DashboardContainer } from '../Dashboard';
+import ClipboardJS from 'clipboard';
+import { toast } from 'react-hot-toast';
+
+const messages = defineMessages({
+  loadingPlaylist: {
+    defaultMessage: 'Loading playlist...',
+    description:
+      'Accessible message for the spinner while loading the playlist in dashboard view.',
+    id: 'component.PlaylistPortability.loadingPlaylist',
+  },
+  playlistTitle: {
+    defaultMessage: 'Belongs to playlist {title} ({id})',
+    description: 'Title for the current playlist.',
+    id: 'component.PlaylistPortability.playlistTitle',
+  },
+  sharedListTitle: {
+    defaultMessage: 'Playlist shared with:',
+    description: '',
+    id: 'component.PlaylistPortability.sharedListTitle',
+  },
+  shareWithPlaylist: {
+    defaultMessage: 'Share with another playlist',
+    description: 'Input label for adding a new portability.',
+    id: 'component.PlaylistPortability.shareWithPlaylist',
+  },
+  shareWithPlaylistDetails: {
+    defaultMessage:
+      'Sharing this playlist will grant access to all this resources.',
+    description: 'Helper text to explain what sharing will do.',
+    id: 'component.PlaylistPortability.shareWithPlaylistDetails',
+  },
+  shareWithPlaylistPlaceholder: {
+    defaultMessage: 'Paste playlist uuid',
+    description: 'Helper text for adding a new portability.',
+    id: 'component.PlaylistPortability.shareWithPlaylistPlaceholder',
+  },
+  updatePlaylistSuccess: {
+    defaultMessage: 'Playlist updated.',
+    description: 'Message when playlist update is successful.',
+    id: 'component.PlaylistPortability.updatePlaylistSuccess',
+  },
+  updatePlaylistFail: {
+    defaultMessage: 'Playlist not updated!',
+    description: 'Message when playlist update failed.',
+    id: 'component.PlaylistPortability.updatePlaylistFail',
+  },
+  removePortability: {
+    defaultMessage: 'Delete sharing with {title}',
+    description: 'Accessible message for removing a portability.',
+    id: 'component.PlaylistPortability.removePortability',
+  },
+  hasPortabilityWith: {
+    defaultMessage: 'Shared with {title}',
+    description:
+      'Accessible message for displaying a portability with a playlist.',
+    id: 'component.PlaylistPortability.hasPortabilityWith',
+  },
+  copied: {
+    defaultMessage: '{text} copied!',
+    description: 'Message displayed when playlist info are copied.',
+    id: 'components.PlaylistPortability.copied',
+  },
+});
+
+interface PlaylistPortabilityProps {
+  object: Video | Document;
+}
+
+export const PlaylistPortability = ({ object }: PlaylistPortabilityProps) => {
```

Comment:
This component is really big and I think can be split in smaller ones. The list itself should be in an other component, maybe you can fnd other part to split.

Reply:
Let me know if you feel more splitting is needed.